src/main/java/bio/terra/tanagra/underlay/entitygroup/CriteriaOccurrence.java

```diff
@@ -85,25 +99,36 @@ public static CriteriaOccurrence fromSerialized(
 
     // Relationships.
     Map<String, Relationship> relationships =
-        Map.of(
-            OCCURRENCE_TO_CRITERIA_RELATIONSHIP_NAME,
-            new Relationship(
+        new HashMap<>(
+            Map.of(
                 OCCURRENCE_TO_CRITERIA_RELATIONSHIP_NAME,
-                occurrenceEntity,
-                criteriaEntity,
-                buildRelationshipFieldList(criteriaEntity)),
-            OCCURRENCE_TO_PRIMARY_RELATIONSHIP_NAME,
-            new Relationship(
+                new Relationship(
+                    OCCURRENCE_TO_CRITERIA_RELATIONSHIP_NAME,
+                    occurrenceEntity,
+                    criteriaEntity,
+                    buildRelationshipFieldList(criteriaEntity)),
                 OCCURRENCE_TO_PRIMARY_RELATIONSHIP_NAME,
-                occurrenceEntity,
-                primaryEntity,
-                Collections.emptyList()),
-            CRITERIA_TO_PRIMARY_RELATIONSHIP_NAME,
-            new Relationship(
+                new Relationship(
+                    OCCURRENCE_TO_PRIMARY_RELATIONSHIP_NAME,
+                    occurrenceEntity,
+                    primaryEntity,
+                    Collections.emptyList()),
                 CRITERIA_TO_PRIMARY_RELATIONSHIP_NAME,
-                criteriaEntity,
-                primaryEntity,
-                buildRelationshipFieldList(criteriaEntity)));
+                new Relationship(
+                    CRITERIA_TO_PRIMARY_RELATIONSHIP_NAME,
+                    criteriaEntity,
+                    primaryEntity,
+                    buildRelationshipFieldList(criteriaEntity))));
+    for (int i = 0; i < occurrenceRelatedEntities.size(); i++) {
```

Comment:
Could this be done via a stream or forEach?

Reply:
Not sure it would be a benefit.. the code uses `i` as well as `occurrenceRelatedEntities.get(i)` so you need both the object and the index in the loop.